src/Resolver/ResolverBase.php

```diff
@@ -94,11 +94,15 @@ public function findPatchesInJson(array $patches): array
                 $temporary_patch_list = [];
 
                 foreach ($patch_defs as $description => $url) {
-                    $patch = new Patch();
-                    $patch->package = $package;
-                    $patch->url = $url;
-                    $patch->description = $description;
-
+                    $patch = [];
+                    if (is_array($url)) {
+                      foreach ($url as $patchdescription => $patchurl) {
+                        $patch[] = $this->getPatches($package, $patchdescription, $patchurl, $description);
+                      }
+                    }
+                    else {
+                      $patch = $this->getPatches($package, $description, $url);
+                    }
```

Comment:
```
if (is_array($url)) {
                      foreach ($url as $patchdescription => $patchurl) {
                        $patch = $this->getPatches($package, $patchdescription, $patchurl, $description);
                        $temporary_patch_list[] = $patch;
                      }
                    }
                    else {
                      $patch = $this->getPatches($package, $description, $url);
                      $temporary_patch_list[] = $patch;
                    }
                    
```
this code should be moved inside foreach and else condition, as $patch is a type of patch object and we are overriding it to convert it into array.

Reply:
also i think we should handle exceptions for wrong urls
`if ($host === NULL) {
          $valid_patches_file = file_exists(realpath($patches_file)) && is_readable(realpath($patches_file));
        }
        else {
          $array = get_headers($patches_file);
          $string = $array[0];

          // 404 for error, 200 for no error
          if(!strpos($string, "200")) {
            return;
          }
        }`

Reply:
also i think we should handle exceptions for wrong urls

```
if ($host === NULL) {
          $valid_patches_file = file_exists(realpath($patches_file)) && is_readable(realpath($patches_file));
        }
        else {
          $array = get_headers($patches_file);
          $string = $array[0];

          if(!strpos($string, "200")) {
            return;
          }
        }
```